packages/esm-patient-registration-app/src/patient-registration/field/id/id-field.test.tsx

```diff
@@ -0,0 +1,220 @@
+    },
+  ];
+  const mockResourcesContextValue = {
+    addressTemplate: [],
+    currentSession: {
+      authenticated: true,
+      sessionId: 'JSESSION',
+      currentProvider: { uuid: 'provider-uuid', identifier: 'PRO-123' },
+    },
+    relationshipTypes: [],
+    identifierTypes: [...mockedIdentifierTypes],
+  } as Resources;
+
+  const openmrsID = {
+    name: 'OpenMRS ID',
+    fieldName: 'openMrsId',
+    required: true,
+    uuid: '05a29f94-c0ed-11e2-94be-8c13b969e334',
+    format: null,
+    isPrimary: true,
+    identifierSources: [
+      {
+        uuid: '691eed12-c0f1-11e2-94be-8c13b969e334',
+        name: 'Generator 1 for OpenMRS ID',
+        autoGenerationOption: {
+          manualEntryEnabled: false,
+          automaticGenerationEnabled: true,
+        },
+      },
+      {
+        uuid: '01af8526-cea4-4175-aa90-340acb411771',
+        name: 'Generator 2 for OpenMRS ID',
+        autoGenerationOption: {
+          manualEntryEnabled: true,
+          automaticGenerationEnabled: true,
+        },
+      },
+    ],
+    autoGenerationSource: null,
+  };
+  (useConfig as jest.Mock).mockImplementation(() => ({
+    defaultPatientIdentifierTypes: ['OpenMRS ID'],
+  }));
+
+  it('should render loading skeleton when identifier types are loading', () => {
+    render(
+      <ResourcesContext.Provider value={[]}>
+        <Formik initialValues={{}} onSubmit={null}>
+          <Form>
+            <PatientRegistrationContext.Provider
+              value={{
+                setFieldValue: jest.fn(),
+                initialFormValues: { identifiers: { openmrsID } },
+                setInitialFormValues: jest.fn(),
+                values: {
+                  identifiers: { openmrsID },
+                },
+              }}>
+              <Identifiers />
+            </PatientRegistrationContext.Provider>
+          </Form>
+        </Formik>
+      </ResourcesContext.Provider>,
+    );
+    expect(screen.getByTestId('loading-skeleton')).toBeInTheDocument();
+  });
+
+  it('should render identifier inputs when identifier types are loaded', () => {
+    render(
+      <ResourcesContext.Provider value={mockResourcesContextValue}>
+        <Formik initialValues={{}} onSubmit={null}>
+          <Form>
+            <PatientRegistrationContext.Provider
+              value={{
+                setFieldValue: jest.fn(),
+                initialFormValues: { identifiers: { openmrsID } },
+                setInitialFormValues: jest.fn(),
+                values: {
+                  identifiers: { openmrsID },
+                },
+              }}>
+              <Identifiers />
+            </PatientRegistrationContext.Provider>
+          </Form>
+        </Formik>
+      </ResourcesContext.Provider>,
+    );
+
+    expect(screen.getByText('Identifiers')).toBeInTheDocument();
+    expect(screen.getByText('Configure')).toBeInTheDocument();
+    expect(screen.getByText('Configure')).toBeEnabled();
```

Comment:
If this is related to a Button, we can use getByRole('button', { name: 'Configure' }). We need to make sure that we are querying the correct element.

(Use getByRole() whenever it is possible.)